apps/condo/domains/meter/schema/RegisterMetersReadingsService.js

```diff
@@ -298,13 +317,36 @@ const RegisterMetersReadingsService = new GQLCustomSchema('RegisterMetersReading
 
                 const resultRows = []
 
+                const meterNumbers = uniq(readings.map(reading => reading.meterNumber.trim()))
+                const meters = await find('Meter', {
+                    organization,
+                    number_in: meterNumbers,
+                    deletedAt: null,
+                })
+
+                const readingsWithValidDates = readings.filter(reading => isDateValid(reading.date))
+                const meterDatePairs = readingsWithValidDates.map(reading => ({
+                    meterNumber: reading.meterNumber.trim(),
+                    date: toISO(reading.date),
```

Comment:
Should you check, that all dates are valid? (`toISO` can return `undefined`, so filter is skipped), or String is not valid date

Reply:
In line 327 we have filter readings with valid dates